connectors/services/base.py

```diff
@@ -59,3 +60,28 @@ def raise_if_spurious(self, exception):
 
         self.errors[0] = errors
         self.errors[1] = first
+
+
+class MultiService:
+    def __init__(self, services):
+        self._services = services
+
+    async def run(self):
+        tasks = [asyncio.create_task(service.run()) for service in self._services]
+
+        done, pending = await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
+
```

Comment:
It will be helpful to add some logs, what tasks complete, and what are pending

Reply:
I'm actually not sure if it's possible - we will get some ids of tasks, but it's hard to correlate task id to a specific action that the task was doing.

We can try to do it in a later PR